Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1842 @@
+
+        Returns:
+            response (Response): API response from Sophos.
+        """
+        url_suffix = f'endpoint/v1/settings/exploit-mitigation/applications/{mitigation_id}'
+        response = self._http_request(method='DELETE', headers=self.headers,
+                                      url_suffix=url_suffix)
+        return response
+
+    def list_detected_exploit(self, page_size: Optional[int], page: Optional[int],
+                              thumbprint_not_in: Optional[str]) -> Dict:
+        """
+        List all detected exploits.
+
+        Args:
+            page_size (int): Size of the page requested.
+            page (int): Number of page to return.
+            thumbprint_not_in (str): Filter out detected exploits with these thumbprints.
+
+        Returns:
+            response (Response): API response from Sophos.
+        """
+        url_suffix = 'endpoint/v1/settings/exploit-mitigation/detected-exploits'
+        params = {'pageSize': page_size, 'page': page, 'thumbprintNotIn': thumbprint_not_in}
+        response = self._http_request(method='GET', headers=self.headers, params=params,
+                                      url_suffix=url_suffix)
+        return response
+
+    def get_detected_exploit(self, detected_exploit_id: str) -> Dict:
+        """
+        Get a single detected exploit.
+
+        Args:
+            detected_exploit_id (str): ID of the exploit to return.
+
+        Returns:
+            response (Response): API response from Sophos.
+        """
+        url_suffix = 'endpoint/v1/settings/exploit-mitigation' \
+                     f'/detected-exploits/{detected_exploit_id}'
+        response = self._http_request(method='GET', headers=self.headers,
+                                      url_suffix=url_suffix)
+        return response
+
+
+def get_client_data(bearer_token: str):
+    """
+    Set the tenant ID required for the API endpoints. Updates the existing self.headers.
+
+    Args:
+        bearer_token (str): A JWT token.
+
+    Returns:
+        headers (dict): Headers for the client.
+        base_url (str): Base URL for the client.
+    """
+    headers = {'Authorization': 'Bearer ' + bearer_token}
+    response = requests.get(headers=headers,
+                            url='https://api.central.sophos.com/whoami/v1')
+    try:
+        response_data = response.json()
+    except (json.JSONDecodeError, DemistoException) as exception:
+        raise DemistoException('Failed getting a whoami response JSON: ', exception) from exception
+    headers.update({'X-Tenant-ID': response_data.get('id')})
+    if not headers.get('X-Tenant-ID'):
+        raise DemistoException('Error retrieving tenant ID.')
+    data_region = response_data.get('apiHosts', {}).get('dataRegion')
+    if not data_region:
+        raise DemistoException(f'Error finding data region. {str(response)}')
+    base_url = f'{data_region}/'
+    return headers, base_url

```

Comment:
is this func needed anymore?

Reply:
Why wouldn't it be?

Reply:
you have it the client, and this one is not being called.

Reply:
oh I might've left it in, my bad